<launch>
	<arg name="speed_linear" default="0.2" />
	<arg name="speed_angular" default="1.0" />
	<arg name="launch_delay" default="0.0" />
	<arg name="wait_for_ready" default="false" />
	<arg name="cmd_vel_topic" default="/miro/control/cmd_vel" />

	<group unless="$(arg wait_for_ready)">
		<rosparam param="wait_for_ready">false</rosparam>
	</group>

	<!-- Teleop to control Miro -->
	<node pkg="joy" type="joy_node" name="joy">
		<rosparam param="joy">/dev/input/js0</rosparam>
		<rosparam param="autorepeat_rate">25</rosparam>
	</node>

	<!-- Collect image data -->
	<node pkg="miro_teach_repeat" type="teleop_joy.py" name="teleop_joy" launch-prefix="bash -c 'sleep $(arg launch_delay); $0 $@'" >
		<rosparam command="load" file="$(find miro_teach_repeat)/config/xbox_one.yaml" />
		<remap from="cmd_vel" to="$(arg cmd_vel_topic)" />
		<rosparam param="scale_linear" subst_value="true">$(arg speed_linear)</rosparam>
		<rosparam param="scale_angular" subst_value="true">$(arg speed_angular)</rosparam>
	</node>
</launch>